rfcs/detecting_sensitive_textual_content/20230309-project_proposal_detecting_sensitive_textual_content.md

```diff
@@ -0,0 +1,379 @@
+   to blurring the images.
+2. It avoids performance issues with relying on low-spec clients to apply
+   Gaussian blurs to potentially several images.
+3. BlurHash hashes have significant future value and all potential API level
+   performance issues can be ameliorated through caching hashes with Redis.
+
+The only significant downside that I think exists is the delay between
+unblurring and displaying the full image due to needing to download the
+full unblurred image. However, I think this is balanced out by the bytes saved
+over the network in _not_ sending images that might potentially never be
+unblurred.
+
+For reviewers: **What are you feelings on this issue and why?**. One alternative
+to going straight for BlurHash, if we wanted to get blurring out the door
+quickly is to treat BlurHash as a future "nice to have" and use CSS blurring in
+the meantime. This would allow us to get a fully working feature slightly more
+quickly. I'm not totally convinced it'd be _that_ much faster though. Like I
+said above, BlurHash looks complicated, but really only in comparison to how
+simple the CSS approach is. Anything would look complicated when compared to
+just adding a CSS filter style to something.
+
+##### Important detail
+
+The success of this project rests on the completion of its technical
+implementation. However, we can add the following events to the API and frontend
+to better understand the usage of the features.
+
+### API
+
+- `SENSITIVE_RESULT_COUNT`
+  - Sent for each query where the search param `include_sensitive_results` is
+    set to `True`. It should include a raw count of the number of results with
+    sensitive textual content or with `mature` set to `True`. Not sent for
+    results where the search params would not include sensitive results.
+
+### Frontend
+
+- `SENSITIVE_RESULTS_TOGGLED`
+  - Sent each time the "include sensitive results" setting is toggled. Include a
+    prop noting if the setting was being toggled on or off.
+- `SENSITIVE_RESULT_UNBLURRED`
+  - Sent each time a result is unblurred. This should include a property
+    denoting whether it on the single result or search results page.
+- `SENSITIVE_RESULT_REBLURRED`
+  - Sent each time a result is reblurred. This should include a property
+    denoting whether it on the single result or search results page.
+- `NO_BLUR_TOGGLED`
+  - Sent each time the "do not blur sensitive results" setting is toggled.
+    Include a prop noting if the setting was being toggled on or off.
+
+## Participants and stakeholders
+
+- Lead: @sarayourfriend
+- Implementation:
+  - @sarayourfriend
+  - TBD
+- Stakeholders:
+  - Team lead: @zackkrida
+
+## Infrastructure
+
+There are no infrastructure changes anticipated for this project.
+
+## Marketing
+
+We can coordinate with marketing to describe the new feature and its motivations
+and celebrate the increased safety for users, especially young people, using
+Openverse.
```

Comment:
I would also emphasize the history Openverse has with educators and students specifically. This change would be huge for them and a great way to reach out to and reestablish ties with those communities. @cogdog ([alan levine](https://www.oeglobal.org/about-us/people-of-oeg/staff/alan-levine/)) is a contributor who comes to mind as someone to help with this outreach. He already does lots of advocating for Openverse.

Reply:
I added a comment in the next commit.